src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -0,0 +1,261 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.Text;
+using Sonarlint;
+using SonarLint.VisualStudio.Integration.Vsix.Helpers;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    ///<summary>
+    /// Tracks SonarLint errors for a specific buffer.
+    ///</summary>
+    /// <remarks>
+    /// <para>The lifespan of this object is tied to the lifespan of the taggers on the view. On creation of the first tagger,
+    /// it starts tracking errors. On the disposal of the last tagger, it shuts down.</para>
+    /// <para>
+    /// See the README.md in this folder for more information
+    /// </para>
+    /// </remarks>
+    internal sealed class TextBufferIssueTracker : IIssueConsumer
```

Comment:
Most of this code was factored out of the old _IssueTagger_ class.
The new version is based very closely on [the equivalent class](https://github.com/Microsoft/VSSDK-Extensibility-Samples/blob/master/ErrorList/C%23/SpellChecker.cs) in the VS Extensibility Error List sample.